datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1175,6 +1180,47 @@ pub(crate) fn from_substrait_type(dt: &substrait::proto::Type) -> Result<DataTyp
     }
 }
 
+fn is_substrait_type_nullable(dtype: &Type) -> Result<bool> {
+    fn is_nullable(nullability: i32) -> bool {
+        nullability != substrait::proto::r#type::Nullability::Required as i32
+    }
+
+    let nullable = match dtype
+        .clone()
+        .kind
+        .ok_or(substrait_datafusion_err!("Type must contain Kind"))?
```

Comment:
```suggestion
    let nullable = match dtype
        .kind
        .as_ref()
        .ok_or_else(|| substrait_datafusion_err!("Type must contain Kind"))?
```
Avoid cloning and creating DataFusionError.